integration/experiment/system_characterization/sys_charac_utils.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+set -e
+set -x
+
+module load geopm-runtime
+
+SWEEP_TYPE="${SWEEP_TYPE?-Set SWEEP_TYPE to either \"CPU\" or \"GPU\"}"
+PROGRAM_NAME="${PROGRAM_NAME?-Set PROGRAM_NAME to the binary name of the characterization application}"
+BINARY_PATH_PLUS_FLAGS="${BINARY_PATH_PLUS_FLAGS?-Set BINARY_PATH_PLUS_FLAGS to the full path of the application binary including the arguments}"
+EMPTY_SWEEP_OUTPUT_DIR="${EMPTY_SWEEP_OUTPUT_DIR?-Set EMPTY_SWEEP_OUTPUT_DIR to the output directory of the characterization sweeps}"
+INIT_CONTROLS_LIST="${EMPTY_SWEEP_OUTPUT_DIR}/init_controls-${HOSTNAME}.lst"
+
+TRIAL_COUNT=3
+CORE_SKIP_LIST="0,1,52,53" # skip first 2 cores on a dual-socket, 52-core/socket system
+
+CPU_LOGICAL_CORE_CNT=$(geopmread -d | grep cpu | awk '{print $2}')
+CPU_PHYSICAL_CORE_CNT=$(geopmread -d | grep core | awk '{print $2}')
+CPU_LOGICAL_CORE_CNT_MINUS_1=$(( ${CPU_LOGICAL_CORE_CNT} - 1 ))
+CPU_PHYSICAL_CORE_CNT_MINUS_4=$(( ${CPU_PHYSICAL_CORE_CNT} - 4 ))
+
+mkdir -p $EMPTY_SWEEP_OUTPUT_DIR
+echo "Writing logs and reports to ${EMPTY_SWEEP_OUTPUT_DIR}"
+
+
+set_rank_bind_list () {
```

Comment:
I'm wondering if it would be better to just hard code the mask list for the different types of sweeps you're doing and avoid/remove this complicated function all together.  This script is already specific Aurora re: skipping cores.  You only run with 1 rank in the GPU case.  We should discuss what the correct number of ranks is in the CPU case.